ironfish/src/utils/array.test.ts

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { ArrayUtils } from './array'
+
+describe('ArrayUtils', () => {
+  it('shuffles array', () => {
+    let items: number[] = []
+
+    for (let i = 0; i <= 10000; ++i) {
+      items.push(i)
+    }
+
+    items = ArrayUtils.shuffle(items)
+
+    const shuffled = items[0] !== 0 && items[500] !== 500 && items[1000] !== 1000
+    expect(shuffled).toBe(true)
```

Comment:
this will randomly fail sometimes (maybe ~3/1000 times?)

i think any assertions we make about random outputs might fail randomly since even very unlikely outcomes are possible

Reply:
oh yeah I did make this better now